### PR TITLE
[WIP] feat: Add PayPal payment configuration

### LINKED ISSUE #333: add paypal

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #333

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 1]
```diff
@@ -531,6 +545,36 @@ class AttendeeFragment : Fragment() {
...
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == PAYPAL_REQUEST_CODE) {
+            if (resultCode == Activity.RESULT_OK) {
+                val paymentConfirmation = data?.getParcelableExtra<PaymentConfirmation>(PaymentActivity.EXTRA_RESULT_CONFIRMATION)
+                if (paymentConfirmation != null) {
+                    val paymentInfo = paymentConfirmation.toJSONObject()
+                    Timber.d(paymentInfo.toString(4))
+                    // TODO: Send token to server and redirect to tickets
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: @iamareebjamal I need help here. I'm getting following response - 
```
{
                        "client": {
                            "environment": "sandbox",
                            "paypal_sdk_version": "2.16.0",
                            "platform": "Android",
                            "product_name": "PayPal-Android-SDK"
                        },
                        "response": {
                            "create_time": "2018-12-30T19:59:20Z",
                            "id": "PAY-0JW58086WS3706240LQUSHEQ",
                            "intent": "sale",
                            "state": "approved"
                        },
                        "response_type": "payment"
                    }
```
And according to api documentation for paypal payment request - 
```
RequestHide
Headers
Accept: application/vnd.api+json
Authorization: JWT <Auth Key>
Body
{
  "data": {
    "attributes": {
      "return_url": "https://localhost:5000/return_url",
      "cancel_url": "https://localhost:5000/cancel_url"
    },
    "type": "paypal-payment"
  }
}
```

How to send confirmation token to the server?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Are you using Braintree SDK?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: No, It's PayPal Android SDK.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: I think Braintree can replace both, Stripe and PayPal.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Will see

**Comment by liveHarshit** [comment]
Issue: https://github.com/fossasia/open-event-server/issues/5961

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt (hunk 1 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: @liveHarshit from what I understand, the server already implements everything it needs, for now, For a Paypal payment, 2 things are needed: A Payer ID and A Payment ID. The Payment ID is generated by the server and I can see that you can get it from PaymentConfirmation above. For Payer ID, please check if you can generate it with the Paypal SDK, probably there should be a Paypal button for the user to log in or some similar method. After you have the Payer ID and the Payment ID, send a Charge object to the server. I think it should work, I will try Paypal SDK for Android to see if I understand it correctly

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: So payer id is the something id of user's PayPal account?

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Yes, it is the ID of who pay the payment, and it is unchanged while for PayPal user email or relevant information can be changed

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: I think we should Braintree, read comments in this question: https://stackoverflow.com/questions/38229929/getting-the-payerid-from-paypal-android-sdk

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: As it is configured on the server, I don't think we can use it on Android. It is configured to accept approval url.
 https://github.com/paypal/PayPal-node-SDK/issues/24

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: let's me spend a bit more time on this and then we can discuss what solution should we go on this tomorrow meeting, besides, I also think Braintree is better as the old documentation is deleted

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: > I also think Braintree is better as the old documentation is deleted

Exactly, that is the problem. PayPal has deleted the documentation related to that and archive the sdk repo.

**Comment by liveHarshit** [comment]
I just found a blog which writes about PayPal server authentication with android: 
@anhanh11001 Can we make payment with `authorization_code` and `client id`: 
I will send the test response soon.

**Pull request closed without merging**